eventuals/expected.h

```diff
@@ -1,224 +1,156 @@
+  // TODO(benh): support any error type that can be "stringified".
+  static_assert(
+      std::disjunction_v<
+          std::is_base_of<std::exception, std::decay_t<E>>,
+          std::is_base_of<std::exception_ptr, std::decay_t<E>>,
+          std::is_same<std::string, std::decay_t<E>>,
+          std::is_same<char*, std::decay_t<E>>>,
+      "An expected must have an error type derived from "
+      "'std::exception' or be string-like to be used as an eventual");
+
+  return Eventual<T>()
+      .template raises<
+          std::conditional_t<
+              std::disjunction_v<
+                  std::is_base_of<std::exception, std::decay_t<E>>,
+                  std::is_base_of<std::exception_ptr, std::decay_t<E>>>,
+              E,
+              std::runtime_error>>()
+      // NOTE: we only care about "start" here because on "stop"
+      // or "fail" we want to propagate that. We don't want to
+      // override a "stop" with our failure because then
+      // downstream eventuals might not stop but instead try and
+      // recover from the error.
+      .start([expected = std::move(expected)](auto& k) mutable {
+        if (expected.has_value()) {
+          return k.Start(std::move(expected.value()));
+        } else {
+          if constexpr (
+              std::disjunction_v<
+                  std::is_base_of<std::exception, std::decay_t<E>>,
+                  std::is_base_of<std::exception_ptr, std::decay_t<E>>>) {
+            return k.Fail(std::move(expected.error()));
+          } else {
+            return k.Fail(std::runtime_error(std::move(expected.error())));
+          }
+        }
+      });
+}
 
 ////////////////////////////////////////////////////////////////////////
 
-template <typename Value_, typename Errors_ = std::tuple<std::exception>>
-class _Expected {
+// Wrapper for 'tl::expected' that allows it to seamlessly be composed
+// with other eventuals.
+//
+// It's currently not possible to not wrap 'tl::expected' without
+// dramatic changes to how we compose eventuals (e.g., we'd likely
+// need to make 'k()' be a free-standing function and 'ValueFrom' and
+// 'ErrorsFrom' be free-standing types).
+//
+// Even if we did that, however, we may still find ourselves wanting
+// to have an "eventuals flavored expected" that is syntactic sugar
+// for a 'std::expected'. For example, 'eventuals::expected<T>' might
+// be an alias for 'std::expected<T, eventuals::Status>' where
+// 'eventuals::Status' is an eventuals specific error type. We might
+// then also have 'eventuals::grpc::expected<T>' be an alias for
+// 'std::expected<T, eventuals::grpc::Status>', and so forth and so
+// on.
+//
+// For now, an 'eventuals::expected<T>' uses a 'std::string' as the
+// default error type to simplify calls to 'make_unexpected("...")'
+// that take strings which is the majority (if not all) of calls.
+template <typename Value_, typename Error_ = std::string>
+class expected : public tl::expected<Value_, Error_> {
```

Comment:
What do you think about adding one more syntax sugar for expected for `std::exception_ptr`, I think it might be useful to the endusers since they are not worried about the error

Reply:
Yes I totally agree! But let's think more about it and figure out what we want it to look like and then do that in a subsequent review.